plugins/modules/docker_config.py

```diff
@@ -292,7 +306,8 @@ def create_config(self):
 
         try:
             if not self.check_mode:
-                config_id = self.client.create_config(self.name, self.data, labels=labels)
+                config_id = self.client.create_config(self.name, self.data,
+                    labels=labels, templating=self.templating)
```

Comment:
```suggestion
                config_id = self.client.create_config(self.name, self.data, labels=labels, templating=self.templating)
```

Reply:
Also this is a breaking change, since this will break every user not using Docker SDK for Python 5.0.3 (the latest version). That parameter has only been added there (https://docker-py.readthedocs.io/en/stable/change-log.html).

This way should work better:
```suggestion
                kwargs = {}
                if self.templating:
                    kwargs['templating'] = self.templating
                config_id = self.client.create_config(self.name, self.data, labels=labels, **kwargs)
```

Also you need to add `option_minimal_versions` to the `AnsibleDockerClient()` call, similar to https://github.com/ansible-collections/community.docker/blob/9a627f45c9480aebf669b67f21fad57f6023f560/plugins/modules/docker_swarm.py#L650-L673, where you specify that `template_driver` needs `docker_py_version='5.0.3', docker_api_version='1.37'` (https://docs.docker.com/engine/api/v1.37/#operation/ConfigList contains it, the API docs for 1.36 do not).